### PR TITLE
Add ground truth stats to quality control pipeline.

### PR DESCRIPTION
**Pull request by cschuet**

### CODE REVIEW

File: jenkins/worker.py [hunk 1]
```diff
@@ -119,6 +137,26 @@ def extract_stats(inp):
   return result
 
 
+def extract_ground_truth_stats(input_text):
+  """Returns a dictionary of ground truth stats."""
+  result = {}
+  parsed = GROUND_TRUTH_STATS_PATTERN.extract(input_text)
```
Comment (SirVer) [inline jenkins/worker.py]: remove empty lines in this function. I python and with the small indent keeping track of blocks is harder and whitespace can hurt readability quite a bit.

File: jenkins/worker.py [hunk 2]
```diff
@@ -197,7 +236,21 @@ def run(self, ros_distro, run_id):
     run_cmd('gsutil cp {}/{}_* gs://cartographer-ci-artifacts/{}/{}/'.format(
         scratch_dir, rosbag_filename, run_id, self.id))
 
-    info = extract_stats(output)
+    # Download ground truth relations file.
+    run_cmd('gsutil cp gs://cartographer-ci-ground-truth/{}/relations.pb '
+            '{}/relations.pb'.format(self.id, scratch_dir))
+
+    # Calculate metrics.
+    output = run_cmd(
```
Comment (SirVer) [inline jenkins/worker.py]: pull out a run_ros_cmd function and absrtact the sourcing away.

File: jenkins/worker.py [hunk 3]
```diff
@@ -119,6 +137,26 @@ def extract_stats(inp):
   return result
 
 
+def extract_ground_truth_stats(input_text):
+  """Returns a dictionary of ground truth stats."""
+  result = {}
+  parsed = GROUND_TRUTH_STATS_PATTERN.extract(input_text)
+
+  result['ground_truth_abs_trans_err'] = float(parsed['abs_trans_err'])
```
Comment (wohe) [inline jenkins/worker.py]: Maybe make this a `for` loop over `abs_trans_err`, ...? Makes it more readable and more obvious you did not introduce typos. You could also extract a constant for this list of names to reuse.

Comment (cschuet) [reply jenkins/worker.py]: Done

**Pull request merged**